include/ui/pixel-display-painted-item.hpp

```diff
@@ -0,0 +1,509 @@
+   */
+  GetColorFunction getColor;
+  /*
+   * \brief returns the color of the pixel at (x,y) fetching the data from the
+   *        prefetched Buffer buffer
+   */
+  GetPixelFromBufferFunction getPixelFromBuffer;
+  /*
+   * \brief returns the color from the color Table at the index index fetching
+   *        the data from the prefetched Buffer buffer
+   */
+  GetColorFromBufferFunction getColorFromBuffer;
+  /*
+   * \brief updates the image wherever the given change in memory modified the
+   *        image
+   */
+  UpdateMemoryFunction updateMemory;
+  /*
+   * \brief updates all pixels of the image
+   */
+  UpdateAllPixelsFunction updateAllPixels;
+  /*
+   * \brief updates all colors of the image
+   */
+  UpdateAllColorsFunction updateAllColors;
+
+  /*
+   * \brief loads a pointer from memory
+   * \param memoryAccess The access to the memory
+   * \param address Address of the pointer
+   * \param indirect ponter is at address : pointer is address
+   * \param cellSize size of a cell in memory
+   * \param pointerSize size of a pointer in Bits
+   * \returns the pointer
+   */
+  static std::size_t loadPointer(Optional<OutputComponent *> memoryAccess,
+                                 std::size_t address,
+                                 bool indirect,
+                                 std::size_t cellSize,
+                                 std::size_t pointerSize);
+  /*
+   * \brief returns the memoryValue from memory at address
+   * \param memoryAccess The access to the memory
+   * \param address address of the value
+   * \param length length of the value in byte
+   * \param defaultLength length of the output if there is no memoryAccess
+   * \returns the memoryValue from memory at address
+   */
+  static MemoryValue getMemoryValueAt(Optional<OutputComponent *> memoryAccess,
+                                      std::size_t address,
+                                      std::size_t length,
+                                      std::size_t defaultLength = 1);
+
+  // RGB:
+  const static GetPixelFunction RGBGetPixel;
+  const static GetColorFunction RGBGetColor;
+  const static GetPixelFromBufferFunction RGBGetPixelFromBuffer;
+  const static GetColorFromBufferFunction RGBGetColorFromBuffer;
+  const static UpdateMemoryFunction RGBUpdateMemory;
+  const static UpdateAllPixelsFunction RGBUpdateAllPixels;
+  const static UpdateAllColorsFunction RGBUpdateAllColors;
+  // Monochrome
+  const static GetPixelFunction MonochromeGetPixel;
+  const static GetColorFunction MonochromeGetColor;
+  const static GetPixelFromBufferFunction MonochromeGetPixelFromBuffer;
+  const static GetColorFromBufferFunction MonochromeGetColorFromBuffer;
+  const static UpdateMemoryFunction MonochromeUpdateMemory;
+  const static UpdateAllPixelsFunction MonochromeUpdateAllPixels;
+  const static UpdateAllColorsFunction MonochromeUpdateAllColors;
+};
+}
+
+class PixelDisplayPaintedItem : public QQuickPaintedItem {
```

Comment:
As Jan says, one class per file. Also, add class level documentation, i.e. why does this class exist, what problem does it solve etc.